### ISSUE TITLE
Parsing broken since February 21 update

### ISSUE THREAD

**Issue by wiedsche** [omni-resources/civ6-save-parser#19]
Creating a new game with current version of civ 6 with all DLCs enabled after February 21 update ist not possible.
Started game with 8 players. 4 human 4 AI. Upload initial turn as usual fails with message only 6 players found, expected 8.

**Comment by mrosack-aa** [comment]
Can you please attach a sample save?  I did a quick test, created a standard map hotseat game with 4 human, 4 ai, and the parser seemed to read that correctly.  Thanks!

**Comment by mrosack** [comment]
Can you please attach a sample save? I did a quick test, created a standard map hotseat game with 4 human, 4 ai, and the parser seemed to read that correctly. Thanks!

**Comment by wiedsche** [comment]
Here is an example file:
[JAYAVARMAN VII. 1 4000 v. Chr..Civ6Save.zip](https://github.com/pydt/civ6-save-parser/files/6054336/JAYAVARMAN.VII.1.4000.v.Chr.Civ6Save.zip)

In the meantime I could find out more. I think it is related to some leaders / civilizations. In that case, INKA and GALLIC are not recognized. 

I wrote a unit test for that, but could not find out yet, where the problem comes from. Somehow the string parsing fails for that civs. `Don't know what this kind of string is...` is set for some properties of that civilizations. And then the actor name is missing (undefined), so it's not added to the list of civilizations.

My test case for that file:
```
describe('Parse February2021 save', () => {
  const buffer = new Buffer(fs.readFileSync('test/saves/february2021.Civ6Save'));
  const parsed = civ6.parse(buffer).parsed;
  const parsedSimple = civ6.parse(buffer, {simple: true}).parsed;

  it('should have 8 civs', () => {
    expect(parsed.CIVS.length).to.equal(8);
    expect(parsedSimple.CIVS.length).to.equal(8);
  });

  it('should have correct data for the 12 civs', () => {
    expect(parsedSimple.CIVS[0].ACTOR_NAME).to.equal('CIVILIZATION_ENGLAND');
    expect(parsedSimple.CIVS[0].LEADER_NAME).to.equal('LEADER_VICTORIA');

    expect(parsedSimple.CIVS[1].ACTOR_NAME).to.equal('CIVILIZATION_KHMER');
    expect(parsedSimple.CIVS[1].LEADER_NAME).to.equal('LEADER_JAYAVARMAN');

    expect(parsedSimple.CIVS[2].ACTOR_NAME).to.equal('CIVILIZATION_GALLIC');
    expect(parsedSimple.CIVS[2].LEADER_NAME).to.equal('LEADER_AMBIORIX');

    expect(parsedSimple.CIVS[3].ACTOR_NAME).to.equal('CIVILIZATION_GRAN_COLOMBIA');
    expect(parsedSimple.CIVS[3].LEADER_NAME).to.equal('LEADER_SIMON_BOLIVAR');

    expect(parsedSimple.CIVS[4].ACTOR_NAME).to.equal('CIVILIZATION_SCYTHIA');
    expect(parsedSimple.CIVS[4].LEADER_NAME).to.equal('LEADER_TOMYRIS');

    expect(parsedSimple.CIVS[5].ACTOR_NAME).to.equal('CIVILIZATION_PERSIA');
    expect(parsedSimple.CIVS[5].LEADER_NAME).to.equal('LEADER_CYRUS');

    expect(parsedSimple.CIVS[6].ACTOR_NAME).to.equal('CIVILIZATION_INKA');
    expect(parsedSimple.CIVS[6].LEADER_NAME).to.equal('LEADER_PACHACUTEC');

    expect(parsedSimple.CIVS[7].ACTOR_NAME).to.equal('CIVILIZATION_AUSTRALIA');
    expect(parsedSimple.CIVS[7].LEADER_NAME).to.equal('LEADER_JOHN_CURTIN');
  });

  it('should have 22 actors', () => {
    expect(parsed.ACTORS.length).to.equal(22);
    expect(parsedSimple.ACTORS.length).to.equal(22);
  });

  it('is game turn 1', () => {
    expect(parsed.GAME_TURN.data).to.equal(1);
    expect(parsedSimple.GAME_TURN).to.equal(1);
  });

  it('has correct game speed', () => {
    expect(parsed.GAME_SPEED.data).to.equal('GAMESPEED_ONLINE');
    expect(parsedSimple.GAME_SPEED).to.equal('GAMESPEED_ONLINE');
  });

  it('has correct map size', () => {
    expect(parsed.MAP_SIZE.data).to.equal('MAPSIZE_STANDARD');
    expect(parsedSimple.MAP_SIZE).to.equal('MAPSIZE_STANDARD');
  });

  it('has correct map file', () => {
    expect(parsed.MAP_FILE.data).to.equal('Pangaea.lua');
    expect(parsedSimple.MAP_FILE).to.equal('Pangaea.lua');
  });

  it('has all players alive', () => {
    for (let i = 0; i < parsed.CIVS.length; i++) {
      expect(parsed.CIVS[i].PLAYER_ALIVE.data).to.equal(true);
      expect(parsedSimple.CIVS[i].PLAYER_ALIVE).to.equal(true);
    }
  });
});
```

**Comment by mrosack** [comment]
This is just some lucky bits causing the parser to go down the wrong path because I don't understand the entire file format and it's misclassifying some data as compressed.  If you comment out lines 308 and 309 it parses fine and all unit tests pass.  I need to think this over a bit and remember the reason why I have that code there, but if I can't think of anything I'll get this updated this weekend.  Thanks!

**Comment by mrosack** [comment]
Finally fixed the array handling code in dae15b6f5719605ec7f322b5f91e5c00e097e9b8 so that it's not flailing trying to figure out data and accidentally classifying it as compressed.
